packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema-v1.ts

```diff
@@ -352,39 +353,44 @@ export async function transformGraphQLSchemaV1(context, options) {
+    const project = await readProjectConfiguration(resourceDir);
 
-  // Check for common errors
-  const directiveMap = collectDirectivesByTypeNames(project.schema);
-  await warnOnAuth(context, directiveMap.types);
-  await searchablePushChecks(context, directiveMap.types, parameters[ResourceConstants.PARAMETERS.AppSyncApiName]);
+    // Check for common errors
+    const directiveMap = collectDirectivesByTypeNames(project.schema);
+    await warnOnAuth(context, directiveMap.types);
+    await searchablePushChecks(context, directiveMap.types, parameters[ResourceConstants.PARAMETERS.AppSyncApiName]);
 
-  await transformerVersionCheck(context, resourceDir, previouslyDeployedBackendDir, resourcesToBeUpdated, directiveMap.directives);
+    await transformerVersionCheck(context, resourceDir, previouslyDeployedBackendDir, resourcesToBeUpdated, directiveMap.directives);
 
-  const transformerListFactory = await getTransformerFactory(context, resourceDir, authConfig);
+    const transformerListFactory = await getTransformerFactory(context, resourceDir, authConfig);
 
-  let searchableTransformerFlag = false;
+    let searchableTransformerFlag = false;
 
-  if (directiveMap.directives.includes('searchable')) {
-    searchableTransformerFlag = true;
-  }
+    if (directiveMap.directives.includes('searchable')) {
+      searchableTransformerFlag = true;
+    }
 
-  const ff = new AmplifyCLIFeatureFlagAdapter();
-  const allowDestructiveUpdates = context?.input?.options?.[DESTRUCTIVE_UPDATES_FLAG] || context?.input?.options?.force;
-  const sanityCheckRulesList = getSanityCheckRules(isNewAppSyncAPI, ff, allowDestructiveUpdates);
+    const ff = new AmplifyCLIFeatureFlagAdapter();
+    const allowDestructiveUpdates = context?.input?.options?.[DESTRUCTIVE_UPDATES_FLAG] || context?.input?.options?.force;
+    const sanityCheckRulesList = getSanityCheckRules(isNewAppSyncAPI, ff, allowDestructiveUpdates);
 
-  const buildConfig = {
-    ...options,
-    buildParameters,
-    projectDirectory: resourceDir,
-    transformersFactory: transformerListFactory,
-    transformersFactoryArgs: [searchableTransformerFlag, storageConfig],
-    rootStackFileName: 'cloudformation-template.json',
-    currentCloudBackendDirectory: previouslyDeployedBackendDir,
-    featureFlags: ff,
-    sanityCheckRules: sanityCheckRulesList,
-  };
-  const transformerOutput = await buildAPIProject(buildConfig);
+    const buildConfig = {
+      ...options,
+      buildParameters,
+      projectDirectory: resourceDir,
+      transformersFactory: transformerListFactory,
+      transformersFactoryArgs: [searchableTransformerFlag, storageConfig],
+      rootStackFileName: 'cloudformation-template.json',
+      currentCloudBackendDirectory: previouslyDeployedBackendDir,
+      featureFlags: ff,
+      sanityCheckRules: sanityCheckRulesList,
+    };
+    transformerOutput = await buildAPIProject(buildConfig);
+  } catch (err) {
+    throw AmplifyGraphQLTransformerErrorConverter.convert(err);
```

Comment:
Isn't the calling method also catching and handling the same thing? 

https://github.com/aws-amplify/amplify-category-api/blob/8b22fb0d11d7dfbcc4e34f803555376477596b1a/packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema.ts#L17-L22

Reply:
Check the stacktrace above, we are directly calling ```transformGraphQLSchemaV2```. 
I guess this is why we are seeing user errors as faults

Reply:
In which place are you directly calling the `transformGraphQLSchemaV2` method?

Reply:
```
    at GraphQLTransform.transform (/local/home/akz/workspace/amplify-category-api/packages/amplify-graphql-transformer-core/lib/transformation/transform.js:200:19)
    at _buildProject (/local/home/akz/workspace/amplify-category-api/packages/amplify-category-api/lib/graphql-transformer/transform-graphql-schema-v2.js:164:39)
    at async buildAPIProject (/local/home/akz/workspace/amplify-category-api/packages/amplify-category-api/lib/graphql-transformer/transform-graphql-schema-v2.js:133:26)
    at async transformGraphQLSchemaV2 (/local/home/akz/workspace/amplify-category-api/packages/amplify-category-api/lib/graphql-transformer/transform-graphql-schema-v2.js:112:29)
    at async Object.run (/local/home/akz/workspace/amplify-category-api/node_modules/@aws-amplify/amplify-provider-awscloudformation/lib/push-resources.js:138:9)
    at async Promise.all (index 0)
    at async providersPush (/local/home/akz/workspace/amplify-category-api/node_modules/amplify-cli-internal/lib/extensions/amplify-helpers/push-resources.js:160:5)
```

Reply:
Thanks to @Amplifiyer . There is one await missing here: [amplify-category-api/packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema.ts](https://github.com/aws-amplify/amplify-category-api/blob/8b22fb0d11d7dfbcc4e34f803555376477596b1a/packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema.ts#L17-L22)

Reply:
Looks like you're calling it from `push-resources`? Are you also calling the V1 transform method from `push-resources`. If yes, maybe we can refactor this logic in the CLI to use the method @Amplifiyer linked. Is that feasible?

Reply:
the logic was missing await condition. I have also updated the PR